lib/rtp_vp9/vp9_payload_descriptor.ex

```diff
@@ -287,61 +264,59 @@ defmodule Membrane.RTP.VP9.PayloadDescriptor do
             | next_dims
           ], rest}}
 
-      _ ->
+      _error ->
         {:error, :malformed_data}
     end
   end
 
-  defp ss_get_widths_heights(<<n_s::3, 1::1, _g::1, _::3>>, rest, count, dimensions)
+  defp ss_get_widths_and_heights(<<n_s::3, 1::1, _g::1, _::3>>, rest, count, dimensions)
        when count == n_s + 1,
        do: {:ok, {dimensions, rest}}
 
-  defp ss_get_widths_heights(_, _, _, _), do: {:error, :malformed_data}
+  defp ss_get_widths_and_heights(_ss_header, _rest, _count, _dimensions),
+    do: {:error, :malformed_data}
 
   defp ss_get_pg_descriptions(<<_n_s::3, _y::1, 1::1, _::3>>, rest) do
     <<n_g, rest::binary()>> = rest
 
-    1..n_g
-    |> Enum.reduce({:ok, {[], rest}}, fn _i, acc ->
-      case acc do
-        {:ok, {accumulated, rest}} ->
-          case ss_get_pg_description(rest) do
-            {:ok, {pg_description, rest}} ->
-              {:ok, {accumulated ++ [pg_description], rest}}
-
-            _ ->
-              {:error, :malformed_data}
-          end
-
-        _ ->
-          {:error, :malformed_data}
-      end
-    end)
+    with {descriptions, {:ok, rest}} <-
+           1..n_g
+           |> Enum.map_reduce({:ok, rest}, fn _i, acc ->
+             with {:ok, rest} <- acc,
+                  {:ok, {pg_description, rest}} <- ss_get_pg_description(rest) do
+               {pg_description, {:ok, rest}}
+             else
+               _error -> {nil, {:error, :malformed_data}}
+             end
+           end) do
```

Comment:
This can be done out of with. You may also try using `Bunch.Enum.try_map_reduce`